<?xml version="1.0" ?>
<launch xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="https://gist.githubusercontent.com/nalt/dfa2abc9d2e3ae4feb82ca5608090387/raw/roslaunch.xsd">
    
    <arg name="hardware_version" default="ned"/>

    <!-- Arguments -->    
    <group ns="niryo_robot">
        <group ns="info">
            <param name="ros_version" type="string" command="rosversion niryo_robot_bringup" />
            <param name="niryo_robot_type" type="string" value="NIRYO_NED" />
        </group>
    </group>

    <rosparam file="$(find niryo_robot_hardware_interface)/config/default.yaml" command="load" />
    <rosparam file="$(find joints_interface)/config/default.yaml" command="load"/>
    <rosparam file="$(find cpu_interface)/config/default.yaml" command="load"/>

    <rosparam file="$(find joints_interface)/config/$(arg hardware_version)/joints_params.yaml" command="load" ns="joints_interface"/>
</launch>
